src/storage/mutate/UpdateVertexProcessor.cpp

```diff
@@ -409,12 +416,30 @@ void UpdateVertexProcessor::process(const cpp2::UpdateVertexRequest& req) {
     VLOG(3) << "Update vertex, spaceId: " << this->spaceId_
             << ", partId: " << partId << ", vId: " << vId;
     CHECK_NOTNULL(kvstore_);
+    switch (checkFilter(partId, vId)) {
+    case 0 : {
+        // do nothing
+        break;
+    }
+    case -1: {
+        // Return OK while filter out
+        // https://github.com/vesoft-inc/nebula/issues/1888
+        this->pushResultCode(cpp2::ErrorCode::SUCCEEDED, partId);
+        this->onFinished();
+        return;
+    }
+    case -2: {
+        this->pushResultCode(cpp2::ErrorCode::E_INVALID_FILTER, partId);
+        this->onFinished();
+        return;
+    }
+    default: {
+        // do nothing
+    }
+    }
     this->kvstore_->asyncAtomicOp(this->spaceId_, partId,
         [partId, vId, this] () -> std::string {
-            if (checkFilter(partId, vId)) {
-                return updateAndWriteBack(partId, vId);
-            }
-            return std::string("");
+            return updateAndWriteBack(partId, vId);
```

Comment:
I think we'd better check the result of `checkFilter` in callback, otherwise we could have consistency problem, because `asyncAtomicOp` is atomic.

Reply:
Thanks.